pkgs/applications/office/zotero-dev/default.nix

```diff
@@ -0,0 +1,400 @@
+
+        buildPhase = ''
+          rm -rf pdf.js
+          cp -Lr ${pdf-worker-pdfjs}/lib/node_modules/pdf.js pdf.js
+        '';
+
+        preInstall = ''
+          mkdir -p $out/lib/node_modules/pdf-worker
+          cp -r node_modules $out/lib/node_modules/pdf-worker/node_modules
+        '';
+      };
+    in
+    buildNpmPackage {
+      pname = "${pname}-client";
+      inherit src version npmFlags NODE_OPTIONS meta;
+      npmDepsHash = "sha256-b9MCHtt4Ewpt/prEMKtzSbLv3xnP2lnhclu4xDh1QGQ=";
+      nativeBuildInputs = [ rsync ];
+
+      postPatch = ''
+        rm -rf resource/SingleFile
+        cp -Lr ${single-file}/lib/node_modules/single-file resource/SingleFile
+
+        rm -rf chrome/content/zotero/xpcom/utilities
+        cp -Lr ${xpcom-utilities}/lib/node_modules/@zotero/utilities chrome/content/zotero/xpcom/utilities
+
+        rm -rf pdf-reader
+        cp -r ${pdf-reader}/lib/node_modules/pdf-reader pdf-reader
+
+        rm -rf pdf-worker
+        cp -r ${pdf-worker}/lib/node_modules/pdf-worker pdf-worker
+
+        chmod +w . -R
+
+        (
+          cd pdf-reader
+          rm -rf pdf.js
+          cp -Lr ${pdf-reader-pdfjs}/lib/node_modules/pdf.js pdf.js
+        )
+        (
+          cd pdf-worker
+          rm -rf pdf.js
+          cp -Lr ${pdf-worker-pdfjs}/lib/node_modules/pdf.js pdf.js
+        )
+
+        rm -rf translators
+        cp -Lr ${translators}/lib/node_modules/translators-check translators
+
+        rm -rf note-editor
+        cp -Lr ${note-editor}/lib/node_modules/zotero-note-editor note-editor
+
+        chmod +w . -R
+
+        find scripts -type f | xargs sed -i 's/npm ci/#npm ci/g'
+        find scripts -type f | xargs sed -i 's/git/#git/g'
+
+        sed -i 's/npm run build/#npm run build/g' scripts/note-editor.js
+      '';
+
+      installPhase = ''
+        mkdir $out
+        cp -r . $out
+      '';
+    };
+
+  zotero-build = stdenv.mkDerivation {
+    pname = "${pname}-build";
+    inherit version meta;
+    src = fetchFromGitHub {
+      owner = "zotero";
+      repo = "zotero-build";
+      rev = "00e854c6588f329b714250e450f4f7f663aa0222";
+      hash = "sha256-Gvt37jObgSQ10GBYjnCLu5XbUAy3oVTkWPvHbhLF+fw=";
+      fetchSubmodules = true;
+    };
+
+    postPatch = ''
+      sed -i -E "/-aL/a '--chmod=Du=rwx'," xpi/build_xpi
+      #sed -i 's/-aL/-L/' xpi/build_xpi
+    '';
+
+    buildInputs = [ python3 rsync perl ];
+
+    buildPhase = ''
+      python3 xpi/build_xpi -s ${zotero-client}/build -c source -m ${rev}
+    '';
+
+    installPhase = ''
+      mkdir $out
+      cp -r . $out
+    '';
+  };
+
+in
+
+stdenv.mkDerivation {
+  inherit pname version meta;
+
+  src = fetchFromGitHub {
+    owner = "zotero";
+    repo = "zotero-standalone-build";
+    rev = "e9ef6bf21d39cc66f1edefdd5b7429bbaf0c5247";
+    hash = "sha256-NcnbqCN6Pti2KuVX79QLrTk2V/3sMxMrbgRzSTluOtM=";
+    fetchSubmodules = true;
+  };
+
+  patches = [ ./fetchxul.patch ];
+
+  postPatch = ''
+    patchShebangs .
```

Comment:
What directory or files need patching? Reducing the amount of files that are reached improves the speed of this command drastically.